utils/xraymanager.go

```diff
@@ -31,3 +35,22 @@ func CreateXrayServiceManagerAndGetVersion(serviceDetails *config.ServerDetails)
 	}
 	return xrayManager, xrayVersion, nil
 }
+
+func SendXscGitInfoRequestIfEnabled(graphScanParams *services.XrayGraphScanParams, xrayManager *xray.XrayServicesManager) (err error) {
+	if graphScanParams.XscVersion, err = xrayManager.XscEnabled(); err != nil {
+		return err
+	}
+	if graphScanParams.XscVersion != "" && graphScanParams.MultiScanId == "" {
+		multiScanId, err := xrayManager.SendXscGitInfoRequest(graphScanParams.XscGitInfoContext)
+		if err != nil {
+			return fmt.Errorf("failed sending Git Info request to XSC service, error: %s ", err.Error())
+		}
+		graphScanParams.MultiScanId = multiScanId
+		log.Debug(fmt.Sprintf("Created xsc git info successfully. multi_scan_id %s", multiScanId))
+		if err = os.Setenv("JF_MSI", multiScanId); err != nil {
+			// Not a fatal error, if not set the scan will not be shown at the XSC UI, should not fail the scan.
+			log.Debug(fmt.Sprintf("failed setting MSI as environment variable. Cause: %s", err.Error()))
+		}
+	}
```

Comment:
```suggestion
    if graphScanParams.XscVersion == "" || graphScanParams.MultiScanId != "" {
        // No Xsc or multiScanId already provided.
        return
    }
    // Generate multiScanId and set it at the scan params.
    multiScanId, err := xrayManager.SendXscGitInfoRequest(graphScanParams.XscGitInfoContext)
	if err != nil {
		return fmt.Errorf("failed sending Git Info request to XSC service, error: %s ", err.Error())
	}
	graphScanParams.MultiScanId = multiScanId
	log.Debug(fmt.Sprintf("Created xsc git info successfully. multi_scan_id %s", multiScanId))
	if err = os.Setenv("JF_MSI", multiScanId); err != nil {
		// Not a fatal error, if not set the scan will not be shown at the XSC UI, should not fail the scan.
		log.Debug(fmt.Sprintf("failed setting MSI as environment variable. Cause: %s", err.Error()))
	}
```
keeping the code "flat" is more readable and easy to maintain and understand.